pkg/appgw/frontend_ipconfigurations.go

```diff
@@ -0,0 +1,36 @@
+package appgw
+
+import (
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+)
+
+// LookupIPConfigurationByType gets the public or private address depending upon privateIP parameter.
+func LookupIPConfigurationByType(frontendIPConfigurations *[]n.ApplicationGatewayFrontendIPConfiguration, privateIP bool) *n.ApplicationGatewayFrontendIPConfiguration {
+	for _, ip := range *frontendIPConfigurations {
+		if ip.ApplicationGatewayFrontendIPConfigurationPropertiesFormat != nil &&
```

Comment:
It looks like one way to simplify this would be to: `if ip.ApplicationGatewayFrontendIPConfigurationPropertiesFormat == nil {continue}`
Another could perhaps be to return a tuple - the list of private IPs and the list of public IPs.
The caller of this function can decide which one they want to use.




			((privateIP && ip.PrivateIPAddress != nil) || 




				(!privateIP && ip.PublicIPAddress != nil)) { 




			return &ip 




		}